### ISSUE TITLE
Support providing response description different from response body description

### ISSUE THREAD

**Issue by jeremy-w** [fastify/fastify-swagger#336]
## 🚀 Feature Proposal

When `responseDescription` is present in the response schema, then move it up to the response's `description` field. If not present, then proceed with the existing behavior of duplicating the `description` to the response `description`.

## Motivation

I want to separately document the response itself vs the response body. This is especially handy when reusing the payload schema.

## Example

```js
        200: {
          responseDescription: 'Successfully authenticated',
          description: 'Always the same response body. TBH I have no clue why it even sends this object in response. The status code is enough. Go figure.',
          type: 'object',
          properties: {
            authorization: {
              type: 'string',
              enum: ['success'],
            },
          },
        }
```

**Comment by mcollina** [comment]
Would you like to send a Pull Request to address this issue? Remember to add unit tests.

**Comment by jeremy-w** [comment]
@mcollina I have this working for Swagger. I still need to do similarly for OpenAPI, then update the docs.

For docs, should I touch anything beyond the README?

**Comment by mcollina** [comment]
> For docs, should I touch anything beyond the README?

No, that's enough

**Comment by climba03003** [comment]
Resolved by #341
